packages/web/hooks/use-transaction-history.ts

```diff
@@ -0,0 +1,70 @@
+import { useMemo } from "react";
+
+import { useStore } from "~/stores";
+import { api } from "~/utils/trpc";
+
+export const useTransactionHistory = ({
+  pageSize = "100",
+  pageNumber = "0",
+}: {
+  pageNumber?: string;
+  pageSize?: string;
+} = {}) => {
+  const { accountStore, transferHistoryStore } = useStore();
+
+  const wallet = accountStore.getWallet(accountStore.osmosisChainId);
+  const recentTransfers = transferHistoryStore.getHistoriesByAccount(
+    wallet?.address || ""
+  );
+
+  const { data: transactionsData, isLoading } =
+    api.edge.transactions.getTransactions.useQuery(
+      {
+        address: wallet?.address || "",
+        page: pageNumber,
+        pageSize,
+      },
+      {
+        enabled: Boolean(wallet?.isWalletConnected && wallet?.address),
+      }
+    );
+
+  const { transactions, hasNextPage } = useMemo(
+    () =>
+      transactionsData ?? {
+        transactions: [],
+        hasNextPage: false,
+      },
+    [transactionsData]
+  );
+
+  const mergedActivity = useMemo(
+    () => [
+      ...transactions.map((tx) => ({
+        ...tx,
+        __type: "transaction" as const,
```

Comment:
Why is type prepended with `__`?

Reply:
There's already a `type` property and since this is a purely internal property, I prepend a `__`